manifests/grub.pp

```diff
@@ -0,0 +1,44 @@
+# === Copyright
+#
+# Copyright 2018, Kumina B.V., Tim Stoop
+# Licensed under the Apache License, Version 2.0 (the "License");
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+
+# == Class: os_hardening::grub
+#
+# Hardens the grub config
+#
+class os_hardening::grub (
+  String  $user                  = 'root',
+  String  $password_hash         = false,
+  Boolean $boot_without_password = true,
+) {
+
+  file { '/etc/grub.d/01_hardening':
+    content => template('os_hardening/grub_hardening.erb'),
+    notify  => Exec['Grub configuration recreate for os_hardening::grub'],
+  }
+
+  if $boot_without_password {
+    # This sets up Grub on Debian Stretch so you can still boot the system without a password
+    exec { 'Keep system bootable without credentials':
```

Comment:
You should better use [file_line](https://forge.puppet.com/puppetlabs/stdlib#file_line) to make changes / substitutions in files

Reply:
I don't think it's possible to do back references with file_line? As I want to leave all other options there without touching them. The docs do not seem to discuss back references either.